### PR TITLE
Feature/1713 - Switch slugify to symfony string

### LINKED ISSUE #1713: Switch cocur/slugify to symfony/string
## Switch cocur/slugify to symfony/string

in order to use more symfony packages and keep this repo using the same dependencies as [admin bundle](https://github.com/sonata-project/SonataAdminBundle/blob/4.x/composer.json#L60C10-L60C24)

I suggest we change [slugify](https://github.com/sonata-project/SonataPageBundle/blob/4.x/composer.json#L25) to `symfony/string`!

in case it makes sense, then I will try to do it :D

### PR DESCRIPTION
**Pull request by eerison**
## Replace  slugify to symfony string

Closes #1713.

## Changelog

```markdown
### Added
- Handle slugs using `Symfony string`

### Deprecated
- Deprecated `slugify` in favor of symfony string
```

### CODE REVIEW

File: src/Entity/PageManager.php [hunk 1]
```diff
@@ -73,44 +74,7 @@ public function createWithDefaults(array $defaults = []): PageInterface
 
     public function fixUrl(PageInterface $page): void
     {
-        if ($page->isInternal()) {
-            $page->setUrl(null); // internal routes do not have any url ...
-
-            return;
```
Comment (VincentLanglet) [inline src/Entity/PageManager.php:79]: if you put this in the handlePageurl, you lose the early return

Comment (eerison) [reply src/Entity/PageManager.php:79]: Ahhhhh I got it now,  then I will keep internal check here and I will move just hybrid condition

Comment (eerison) [reply src/Entity/PageManager.php:79]: Done

File: src/Entity/PageManager.php [hunk 2]
```diff
@@ -35,7 +37,7 @@ final class PageManager extends BaseEntityManager implements PageManagerInterfac
     public function __construct(
         string $class,
         ManagerRegistry $registry,
-        private SlugifyInterface $slugify,
+        private SlugifyInterface|SluggerInterface $slugger,
```
Comment (VincentLanglet) [inline src/Entity/PageManager.php:40]: You need to add a deprecation if the slugger passed is not instance of SluggerInterface.

And a next major comment to reduce the param to sluggerInterface.

**Review by VincentLanglet — CHANGES_REQUESTED** [review]
With this last change we should be good. Thanks

File: src/Entity/PageManager.php [hunk 3]
```diff
@@ -168,4 +139,55 @@ public function getHybridPages(SiteInterface $site): array
...
+            $url = $page->getCustomUrl() ?? '';
+            $page->setUrl('/'.ltrim($url, '/'));
+
+            return;
+        }
+
+        $slug = $page->getSlug();
+
+        // NEXT_MAJOR: Remove Slugify support
+        if (null === $slug && $this->slugger instanceof Slugify) {
```
Comment (VincentLanglet) [inline src/Entity/PageManager.php:163]:
```suggestion
        if (null === $slug && $this->slugger instanceof SlugifyInterface) {
```
no?

and might be better to write
```suggestion
        if (null === $slug && !$this->slugger instanceof SluggerInterface) {
```

File: src/Entity/PageManager.php [hunk 4]
```diff
@@ -168,4 +139,55 @@ public function getHybridPages(SiteInterface $site): array
...
+            @trigger_error(sprintf(
+                '%s is deprecated since version 4.8.0 and will be removed in 5.0, use %s instead of.',
+                Slugify::class,
```
Comment (VincentLanglet) [inline src/Entity/PageManager.php:166]:
```suggestion
                SlugifyInterface::class,
```

**Pull request closed without merging**